pkg/serviceinfo/serviceinfo.go

```diff
@@ -179,6 +182,11 @@ func (m methodInfo) StreamingMode() StreamingMode {
 	return m.streamingMode
 }
 
+type CodecInfo interface {
+	GetMessageReaderWriter() interface{}
```

Comment:
Move the definition to the generic package, it doesn't related to serviceinfo?

Reply:
Maybe you can move the definition of `ServiceName() string` to the `generic.Generic` interface.

Reply:
And then remove `GetIDLServiceName() string` from CodecInfo interface{}?

Reply:
Actually, I don't think we need the `CodecInfo` interface, just pass the `Generic` interface directly, and the `Generic` interface implements `ServiceName() string` and `GetMessageReaderWriter() interface{}`.

Reply:
oh that's very true. I used to write codes which CodecInfo was accessed from outside of generic package (which was wrong), and after fixing that, I forgot to move the definition. I'll fix it.